File: java/wecker/src/main/java/org/example/Wecker.java
package org.example;

import java.time.Duration;
import java.time.ZonedDateTime;

public class Wecker {
    public WeckerZustand IstGestoppt() {
        return WeckerZustand.Gestoppt;
    }

    public WeckerZustand IstGestartet() {
        return WeckerZustand.Geatartet;
    }

    public Duration RestzeitBerechnen(ZonedDateTime weckzeit, ZonedDateTime uhrzeit) {
        return Duration.between(uhrzeit, weckzeit);
    }

    public void WennRestZeitAbgelaufen(Duration restzeit, Runnable onAlarm) {
        if(restzeit.getSeconds() <= 0) {
            onAlarm.run();
        }
    }
}

File: csharp/wecker_console/wecker/wecker/Wecker.cs
using System;

namespace wecker
{
    public static class Wecker
    {
        public static TimeSpan Restzeit_berechnen(in DateTime uhrzeit, in DateTime weckzeit) {
            return weckzeit - uhrzeit;
        }

        public static bool Restzeit_ist_abgelaufen(TimeSpan restzeit) {
            return restzeit <= new TimeSpan();
        }
    }
}